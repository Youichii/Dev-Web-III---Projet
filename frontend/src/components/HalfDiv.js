const HalfDiv = ({divpos, link, title}) => {  
    return (
       
        <div className={ divpos }> 
            {link}             
        </div>
    )
}

HalfDiv.defaultProps = {
    divpos: 'divleft',  //define a default title for the header
    link: '<div></div>'
}


export default HalfDiv

