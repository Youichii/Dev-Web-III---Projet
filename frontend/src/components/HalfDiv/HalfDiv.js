const HalfDiv = ({divpos, link, title}) => {  
    return (
       
        <div data-testid="halfdiv" className={ divpos }> 
            {link}             
        </div>
    )
}

HalfDiv.defaultProps = {
    divpos: 'divleft',  //define a default title for the header
    link: '<div></div>'
}


export default HalfDiv

