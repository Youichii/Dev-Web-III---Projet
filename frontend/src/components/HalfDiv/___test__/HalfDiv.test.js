import React from 'react';
import ReactDOM from 'react-dom';
import HalfDiv from '../HalfDiv';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<HalfDiv />, div)
})

it("apply the right classname for the div", () => {
    const {getByTestId} = render(<HalfDiv divpos="gauche"/>);
    expect(getByTestId('halfdiv')).toHaveClass("gauche")
})

it("Creates snapeshot with same default props", () => {
    const tree = renderer.create(<HalfDiv  divpos='divleft' link='<div></div>'/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot 1 to see default props", () => {
    const tree = renderer.create(<HalfDiv/>).toJSON();
    expect(tree).toMatchSnapshot();
})