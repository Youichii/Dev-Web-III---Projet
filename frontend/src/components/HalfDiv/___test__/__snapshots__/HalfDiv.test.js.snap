// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates snapeshot with same default props 1`] = `
<div
  className="divleft"
  data-testid="halfdiv"
>
  &lt;div&gt;&lt;/div&gt;
</div>
`;

exports[`matches snapshot 1 to see default props 1`] = `
<div
  className="divleft"
  data-testid="halfdiv"
>
  &lt;div&gt;&lt;/div&gt;
</div>
`;
